forward/events.go

```diff
@@ -0,0 +1,64 @@
+package forward
+
+// An EventListener handles events given to it by the SWIM node. HandleEvent should be thread safe.
+type EventListener interface {
+	HandleEvent(interface{})
```

Comment:
In that case I'd rather use the EventListener that is already in the `events` package. But since `swim` defined its own listener as well I opted for this.

I'm not really keen on passing it as a `func` in this case, especially since the `EventListener` is passed around throughout the project already. Rather keep 1 way of passing them around.